src/controllers/userController.ts

```diff
@@ -0,0 +1,23 @@
+import { FastifyRequest, FastifyReply } from 'fastify';
+
+interface ReqUser {
+    name: string;
+    email: string;
+    password: string;
+    birthDate: Date;
+}
+
+interface ResUser{
+    id: string;
+    name: string;
+    email: string;
+    password: string;
+    birthDate: Date;
+}
```

Comment:
```suggestion
interface ResUser extends ReqUser {
    id: string;
}
```

Assim você não precisa "re-declarar" os outros campos.